src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -188,12 +210,38 @@ void EIRPlantLoopHeatPump::setOperatingFlowRatesWSHP(EnergyPlusData &state)
                                                     this->sourceSideMassFlowRate);
         // Set flows if the heat pump is running
     } else { // the heat pump must run
-        this->loadSideMassFlowRate = this->loadSideDesignMassFlowRate;
-        this->sourceSideMassFlowRate = this->sourceSideDesignMassFlowRate;
-        PlantUtilities::SetComponentFlowRate(
-            state, this->loadSideMassFlowRate, this->loadSideNodes.inlet, this->loadSideNodes.outlet, this->loadSidePlantLoc);
-        PlantUtilities::SetComponentFlowRate(
-            state, this->sourceSideMassFlowRate, this->sourceSideNodes.inlet, this->sourceSideNodes.outlet, this->sourceSidePlantLoc);
+        // apply min/max operating limits based on source side entering fluid temperature
+        if (this->minSourceTempLimit > this->sourceSideInletTemp || this->maxSourceTempLimit < this->sourceSideInletTemp) {
+            this->loadSideMassFlowRate = (this->loadSideIsPlantInlet) ? state.dataLoopNodes->Node(this->loadSideNodes.inlet).MassFlowRate : 0.0;
+            this->sourceSideMassFlowRate =
+                (this->sourceSideIsPlantOutlet) ? state.dataLoopNodes->Node(this->sourceSideNodes.inlet).MassFlowRate : 0.0;
+            this->running = false;
+        } else {
+            this->loadSideMassFlowRate =
+                (this->loadSideIsPlantInlet) ? state.dataLoopNodes->Node(this->loadSideNodes.inlet).MassFlowRate : this->loadSideDesignMassFlowRate;
+            this->sourceSideMassFlowRate = (this->sourceSideIsPlantOutlet) ? state.dataLoopNodes->Node(this->sourceSideNodes.inlet).MassFlowRate
+                                                                           : this->sourceSideDesignMassFlowRate;
+
```

Comment:
The above was meant to identify the HR HP. The `= (this->sourceSideIsPlantOutlet) ?` is shutting down the chiller where loadSideIsPlantInlet = true and sourceSideIsPlantOutput = false. I changed this to:

            this->loadSideMassFlowRate = (this->loadSideIsPlantInlet && this->sourceSideIsPlantOutlet)
                                             ? state.dataLoopNodes->Node(this->loadSideNodes.inlet).MassFlowRate
                                             : this->loadSideDesignMassFlowRate;
            this->sourceSideMassFlowRate = (this->loadSideIsPlantInlet && this->sourceSideIsPlantOutlet)
                                               ? state.dataLoopNodes->Node(this->sourceSideNodes.inlet).MassFlowRate
                                               : this->sourceSideDesignMassFlowRate;

and now the example file PlantLoopHeatPump_EIR_WaterSource runs again. I should change this to a single bool that is true if loadSideIsPlantInlet = true and sourceSideIsPlantOutput = true.